### ISSUE TITLE
tsql: Unicode (e.g. Cyrillic) characters in @parameter names break the parser

### ISSUE THREAD

**Issue by msergein** [sqlfluff/sqlfluff#7806]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

In the `tsql` dialect, `@parameter` names that contain Unicode letters
(Cyrillic, Greek, accented Latin, CJK, …) are not parsed. SQLFluff fails
the entire enclosing batch as one big unparsable section starting at line
1, even though the parameter is a perfectly valid T-SQL identifier in
SQL Server.

T-SQL allows Unicode letters in regular identifiers (and therefore in
parameter names), since SQL Server uses Unicode identifier rules:

> A regular identifier ... must comply with rules from the Unicode 3.2
> standard. ... Letters, as defined in the Unicode Standard 3.2.

Reference (identifier rules):
https://learn.microsoft.com/en-us/sql/relational-databases/databases/database-identifiers

Reference (parameter name rules — same as identifier rules):
https://learn.microsoft.com/en-us/sql/t-sql/statements/declare-local-variable-transact-sql

### Root cause (from reading the source) by claude code

In `src/sqlfluff/dialects/dialect_tsql.py` (line ~578):

```python
ParameterNameSegment=RegexParser(
    r"@(?!@)[A-Za-z0-9_@$#]+", CodeSegment, type="parameter"
),
```

The character class `[A-Za-z0-9_@$#]+` is ASCII-only and so does not match
any Unicode letter that T-SQL identifier rules permit.

Suggested fix: broaden the regex to allow Unicode letters/digits/marks. A
conservative replacement that still excludes whitespace and operators
would be something like:

```python
ParameterNameSegment=RegexParser(
    r"@(?!@)[\w@$#]+",         # \w is Unicode-aware in Python re
    CodeSegment, type="parameter"
),
```

…or, if a stricter character set is preferred, an explicit Unicode class
matching SQL Server's identifier rules (`L`, `Mn`, `Mc`, `Nd`, plus `_`,
`@`, `$`, `#`):

```python
ParameterNameSegment=RegexParser(
    r"@(?!@)[\p{L}\p{Mn}\p{Mc}\p{Nd}_@$#]+",
    CodeSegment, type="parameter"
),
```

(`regex` package or `re` with the appropriate flag, depending on the
project's preferred dependency).

A YAML test fixture under
`test/fixtures/dialects/tsql/unicode_parameter_names.{sql,yml}` would
cover regression.

The same issue likely affects `NakedIdentifierSegment` and other
identifier-matching grammars in this dialect — worth a once-over while
fixing parameters.

### Expected Behaviour

Parameter names containing Unicode letters should parse cleanly. For
example:

```sql
DECLARE @Дата DATETIME2
```

…should parse

### Observed Behaviour

```
$ cat repro.sql
DECLARE @Дата DATETIME2

$ sqlfluff parse --dialect tsql repro.sql
==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE FUNCTION
                       | [dbo].[fn_test]\n(\n...'
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
```

### How to reproduce

1. Save the snippet above as `repro.sql` (UTF-8 encoding).
2. Run `sqlfluff parse --dialect tsql repro.sql`.
3. Observe the unparsable section at line 1.
4. Replace `@Дата` with `@Date` — the file parses with no PRS.

### Dialect

tsql

### Version

4.1.0
Python 3.14.4 (Windows)

### Configuration

```
[sqlfluff]
dialect = tsql
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
